client/my-sites/email/email-management/home/email-plan-mailboxes-list.jsx

```diff
@@ -139,24 +276,31 @@ function EmailPlanMailboxesList( { mailboxes, isLoadingEmails } ) {
 
 		return (
 			<MailboxListItem key={ mailbox.mailbox } isError={ !! warning }>
-				<div className="email-plan-mailboxes-list__mailbox-list-item-main">
-					<div>
-						<MaterialIcon icon="email" />
-						<span>
-							{ mailbox.mailbox }@{ mailbox.domain }
-						</span>
+				<div className="email-plan-mailboxes-list__mailbox-list-item-email">
+					<div className="email-plan-mailboxes-list__mailbox-list-item-main">
+						<div>
+							<MaterialIcon icon="email" />
+							<span>
+								{ mailbox.mailbox }@{ mailbox.domain }
+							</span>
+						</div>
+						{ getSecondaryContentForMailbox( mailbox ) }
 					</div>
-					{ getSecondaryContentForMailbox( mailbox ) }
+					{ isEmailUserAdmin( mailbox ) && (
+						<Badge type="info">
+							{ translate( 'Admin', {
+								comment: 'Email user role displayed as a badge',
+							} ) }
+						</Badge>
+					) }
+					{ warning }
+					{ action }
 				</div>
-				{ isEmailUserAdmin( mailbox ) && (
-					<Badge type="info">
-						{ translate( 'Admin', {
-							comment: 'Email user role displayed as a badge',
-						} ) }
-					</Badge>
+				{ hasTitanMailWithUs( domain ) ? (
+					<TitanActionMenu />
+				) : (
+					<GmailActionMenu domainName={ domain.name } />
 				) }
```

Comment:
_Warning: I had a lot of thoughts that got dumped in this comment!_

This component is also used to show email forwards, so I think its going to be simpler to push this logic into a separate function, either a generic/local component (i.e. `const ActionMenu = ( { domain } ) => {}`), or as a helper function (i.e. `const getActionMenu = ( domain ) => {}`), and then drop that into this main JSX block.

I am also thinking that it may be simpler to have a single location where we add an `EllipsisMenu` -- we can then use helper functions to add the relevant menu items, and have only one "menu" component.

I am thinking the code could look something like this:
```js
const TitanMenuItems = () => {
	const translate = useTranslate();

	return (
		<>
			<PopoverMenuItem
				href="https://wp.titan.email/mail/"
				className="email-plan-mailboxes-list__mailbox-action-menu-item"
				isExternalLink={ true }
			>
				<img src={ titanMailIcon } alt={ translate( 'Titan Mail icon' ) } />
				{ translate( 'Mail', {
					comment: 'This refers to the Email application (i.e. the webmail) of Titan',
				} ) }
			</PopoverMenuItem>
			<PopoverMenuItem
				href="https://wp.titan.email/calendar/"
				className="email-plan-mailboxes-list__mailbox-action-menu-item"
				isExternalLink={ true }
			>
				<img src={ titanCalendarIcon } alt={ translate( 'Titan Calendar icon' ) } />
				{ translate( 'Calendar', {
					comment: 'This refers to the Calendar application of Titan',
				} ) }
			</PopoverMenuItem>
			<PopoverMenuItem
				href="https://wp.titan.email/contacts/"
				className="email-plan-mailboxes-list__mailbox-action-menu-item"
				isExternalLink={ true }
			>
				<img src={ titanContactsIcon } alt={ translate( 'Titan Contacts icon' ) } />
				{ translate( 'Contacts', {
					comment: 'This refers to the Contacts application of Titan',
				} ) }
			</PopoverMenuItem>
		</>
	);
};

const GoogleMenuItems = ( { domain } ) => {
   // ...
};

const getMenuItems = ( domain ) => {
	if ( hasTitanMailWithUs( domain ) ) {
		return <TitanMenuItems />;
	}

	if ( hasGSuiteWithUs( domain ) ) {
		return <GoogleMenuItems domain={ domain } />;
	}

	// TODO: Add email forwarding options or remove them
	return null;
};

const ActionMenu = ( { domain } ) => {
	const actionMenuItems = getMenuItems( domain );

	if ( ! actionMenuItems ) {
		return null;
	}

	return (
		<EllipsisMenu
			className="email-plan-mailboxes-list__mailbox-action-menu"
			popoverClassName="email-plan-mailboxes-list__mailbox-action-menu-popover"
		>
			{ actionMenuItems }
		</EllipsisMenu>
	);
};
```

It may not be for this PR, but all the `PopoverMenuItem` boilerplate for the menu items also bothers me. It may be worth defining the data here via arrays and then mapping that to components. e.g.
```js
const getTitanMenuItems = ( translate ) => {
	return [
		{	
			href: "https://wp.titan.email/mail/",
			image: titanMailIcon,
			imageAltText: translate( 'Titan Mail icon' ),
			title: translate( 'Mail', {
				comment: 'This refers to the Email application (i.e. the webmail) of Titan',
			} ),
		},
		{
			href: "https://wp.titan.email/calendar/",
			image: titanCalendarIcon,
			imageAltText: translate( 'Titan Calendar icon' ),
			title: translate( 'Calendar', {
				comment: 'This refers to the Calendar application of Titan',
			} ),
		},
		{
			href: "https://wp.titan.email/contacts/",
			image: titanContactsIcon,
			imageAltText: translate( 'Titan Contacts icon' ),
			title: translate( 'Contacts', {
				comment: 'This refers to the Contacts application of Titan',
			} ),
		}
	];
};

const getMenuItems = ( domain, translate ) => {
	if ( hasTitanMailWithUs( domain ) ) {
		return getTitanMenuItems( translate );
	}
	// ...
};

const ActionMenu = ( { domain } ) {
	const translate = useTranslate();
	const menuItems = getMenuItems( domain, translate );
	if ( ! menuItems ) {
		return null;
	}
	return (
		<EllipsisMenu>
			{ menuItems.map( function( menuItem ) {
				const { image, imageAltText, href,  title } = menuItem;

				// We likely need a key here -- might be best to use the `href`
				return (
					<PopoverMenuItem
						className="email-plan-mailboxes-list__mailbox-action-menu-item"
						isExternalLink
						href={ href }
					>
						<img src={ image } alt={ imageAltText } />
						{ title }
					</PopoverMenuItem>
				);
			} ) }
		</EllipsisMenu>
	);
};
```

Reply:
Created a mappable array out of the menu items as you've suggested.

I needed to understand more about "showing email forwards". Is it something I missed from the designs?

Reply:
I think so -- you can take a look at #52750, especially [this comment](https://github.com/Automattic/wp-calypso/pull/52750#issuecomment-840593024) for a screenshot and some discussion.

I am wondering whether these designs were only in the Figma and not in the post.